nixos/modules/services/web-apps/gancio.nix

```
@@ -0,0 +1,271 @@
+    };
+
+    listenPort = lib.mkOption {
+      type = lib.types.port;
+      default = 13120;
+      description = ''
+        Port number of the gancio server to listen on.
+      '';
+    };
+
+    dbDialect = lib.mkOption {
+      type = lib.types.enum [
+        "sqlite"
+        "postgres"
+      ];
+      default = "sqlite";
+      description = ''
+        The database dialect to use
+      '';
+    };
+
+    logLevel = lib.mkOption {
+      description = "Gancio log level.";
+      type = lib.types.enum [
+        "debug"
+        "info"
+        "warning"
+        "error"
+      ];
+      default = "info";
+    };
+
+
+    settings = lib.mkOption rec {
+      type = json.type;
+      apply = lib.recursiveUpdate default;
+      default = {
+        baseurl = "http${
+          lib.optionalString config.services.nginx.virtualHosts."${cfg.hostName}".enableACME "s"
+        }://${cfg.hostName}";
+        server = {
+          host = cfg.listenHost;
+          port = cfg.listenPort;
+        };
+        log_level = cfg.logLevel;
+        log_path = "/var/log/${cfg.user}";
+        db =
+          {
+            dialect = cfg.dbDialect;
+          }
+          // lib.optionalAttrs (cfg.dbDialect == "sqlite") { storage = "/var/lib/${cfg.user}/db.sqlite"; }
+          // lib.optionalAttrs (cfg.dbDialect == "postgres") {
+            host = "/run/postgresql";
+            database = cfg.user;
+          };
+        user_locale = "/var/lib/${cfg.user}/user_locale";
+        upload_path = "/var/lib/${cfg.user}/uploads";
+      };
+      defaultText = lib.literalExpression ''
+        {
+          baseurl = "http''${
+            lib.optionalString config.services.nginx.virtualHosts."''${cfg.hostName}".enableACME "s"
+          }://''${cfg.hostName}";
+          server = {
+            host = cfg.listenHost;
+            port = cfg.listenPort;
+          };
+          log_level = cfg.logLevel;
+          log_path = "/var/log/''${cfg.user}";
+          db =
+            {
+              dialect = cfg.dbDialect;
+            }
+            // lib.optionalAttrs (cfg.dbDialect == "sqlite") { storage = "/var/lib/''${cfg.user}/db.sqlite"; }
+            // lib.optionalAttrs (cfg.dbDialect == "postgres") {
+              host = "/run/postgresql";
+              database = cfg.user;
+            };
+          user_locale = "/var/lib/''${cfg.user}/user_locale";
+          upload_path = "/var/lib/''${cfg.user}/uploads";
+        }
+      '';
+      example = {
+        db = {
+          dialectOptions.autoJsonMap = false;
+        };
+      };
+      description = ''
+        Configuration for Gancio, see <https://gancio.org/install/config> for supported values.
+      '';
+    };
+
+    userLocale = lib.mkOption {
+      type = with lib.types; attrsOf (attrsOf (attrsOf str));
+      default = { };
+      example = {
+        en = {
+          register = {
+            description = "My new registration page description";
+          };
+        };
```

```suggestion
        en.register.description = "My new registration page description";
```
we can compact the docs a bit here